File: misc/count.pl
my %counter;

while (my $line = <> ) {
    chomp $line;
    foreach my $str (split /\s+/, $line) {
        $counter{$str}++;
    }
}

foreach my $word (sort keys %counter) {
    printf "%s : %s\n", $word, $counter{$word};
}

File: misc/count.py
import sys

def count():

    counter = {}

    for line in sys.stdin:
        for word in line.split():
            item = counter.get(word, 0)
            counter[word] = item + 1


    for word, total in sorted(counter.items(), key=lambda elem : elem[0]):
        print(word, ':', total)


if __name__ == '__main__':
    count()
